poetry/console/application.py

```diff
@@ -137,6 +137,21 @@ def create_io(
     ) -> IO:
         io = super().create_io(input, output, error_output)
 
+        # fmt: off
+        # Remove when support for Python 3.6 is removed
+        # https://github.com/python-poetry/poetry/issues/3412
+        import sys
+        if (
+            sys.version_info[:2] <= (3, 6) and
+            hasattr(io.output, "_stream") and
+            hasattr(io.output._stream, "buffer")
+        ):
+            import io as io_
+            io.output._stream = io_.TextIOWrapper(
+                io.output._stream.buffer, encoding='utf-8'
+            )
+        # fmt: on
```

Comment:
I would rather let black do its thing than introduce an exception. Adding 2 lines to remove 2 does not seem like a good gain.

Also note there is a `PY36` constant in `poetry.utils._compat`.

Reply:
The point wasn't to reduce line count, the point was to avoid making it seem like `import sys` was unrelated to the `if` block. But sure, I can do that.

The `PY36` constant checks if it's at least 3.6, so that won't work. But I can do `not PY37`. Also, it's `poetry.core.utils._compat`